src/files/files-list/FilesList.js

```diff
@@ -305,22 +310,24 @@ export class FilesList extends React.Component {
           </div>
           <div className='pa2' style={{ width: '2.5rem' }} />
         </header>
-        { upperDir &&
-          <File
-            ref={r => { this.filesRefs['..'] = r }}
-            onNavigate={() => this.props.onNavigate(upperDir.path)}
-            onInspect={() => this.props.onInspect([upperDir])}
-            onAddFiles={this.props.onAddFiles}
-            onMove={this.move}
-            setIsDragging={this.isDragging}
-            translucent={isDragging || (isOver && canDrop)}
-            name='..'
-            focused={this.state.focused === '..'}
-            cantDrag
-            cantSelect
-            {...upperDir} />
-        }
-        {this.files}
+        <div className={wrapperAnimClass}>
```

Comment:
Let's extract this to a component! Could we have something like

```jsx
<LoadingAnimation loading={filesIsFetchin}>
  <File .../>
</LoadingAnimation>
```

Reply:
We can, but it won't just work out of the box by itself, because the parent container must have some logic and specific css classes for it to work.

Reply:
@olizilla can you check commit https://github.com/ipfs-shipyard/ipfs-webui/pull/949/commits/42a2304b6f0bf3e7bac9dff76359c9c2a6a04077 and see what you think?

Do you know a way of decoupling and keeping all the logic needed inside the `<LoadingAnimation>`, without having the `<FilesList>` component to have to deal with something?

Reply:
How about https://github.com/ipfs-shipyard/ipfs-webui/pull/949/commits/d6722ac2a1ab23e6ea33f8799a85cacfa0a4a21f

Reply:
Love it ❤️